File: Openapi_test/app/src/main/java/com/example/openapi_test/MainActivity.java
package com.example.openapi_test;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import org.w3c.dom.Text;
import org.xmlpull.v1.XmlPullParser;
import org.xmlpull.v1.XmlPullParserFactory;

import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.URL;
import java.net.URLEncoder;

public class MainActivity extends AppCompatActivity {

    EditText edit;
    TextView text;

    // 자신의 서비스키 입력
    String key = "LHSwFqTiNHfQEQw4PRhQhq4kY40LwyN9%2Fx4Q9NpOWQr8Dqqql2i630psVHxI4l7Ha37sT292A6M5SR1NCb41rQ%3D%3D";
    //String address = "http://openapi.forest.go.kr/openapi/service/trailInfoService/getforeststoryservice";
    String data;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        edit = (EditText)findViewById(R.id.edit);
        text = (TextView)findViewById(R.id.result);
    }

    public void mOnClick(View v) {
        switch (v.getId()) {
            case R.id.button:
                new Thread(new Runnable() {
                    @Override
                    public void run() {
                        data = getXmlData();

                        runOnUiThread(new Runnable() {
                            @Override
                            public void run() {
                                text.setText(data);
                            }
                        });
                    }
                }).start();
                break;
        }
    }

    String getXmlData(){
        StringBuffer buffer = new StringBuffer();

        String str = edit.getText().toString();
        String location = URLEncoder.encode(str);

        String queryUrl = "http://openapi.forest.go.kr/openapi/service/trailInfoService/getforeststoryservice"
                + "?serviceKey=" + key
                + "&mntnAdd=" + location;

        try {
            URL url = new URL(queryUrl);
            InputStream is = url.openStream();

            XmlPullParserFactory factory = XmlPullParserFactory.newInstance();
            XmlPullParser xpp = factory.newPullParser();
            xpp.setInput(new InputStreamReader(is, "UTF-8"));

            String tag;

            xpp.next();
            int eventType = xpp.getEventType();

            while(eventType != XmlPullParser.END_DOCUMENT) {
                switch (eventType) {
                    case XmlPullParser.START_DOCUMENT:
                        buffer.append("Parsing Start..\n\n");
                        break;

                    case XmlPullParser.START_TAG:
                        tag = xpp.getName(); // Get tag name

                        if (tag.equals("item"));  // First search result
                        else if (tag.equals("mntninfodtlinfocont")){
                            buffer.append("Detail_info : ");
                            xpp.next();
                            buffer.append(xpp.getText()); // Add to StringBuffer from text of "mntninfodtlinfocont" element
                            buffer.append("\n");  // Add line-break
                        }
                        else if (tag.equals("mntnnm")){
                            buffer.append("Mountain_Name : ");
                            xpp.next();
                            buffer.append(xpp.getText()); // Add to StringBuffer from text of "mntnnm" element
                            buffer.append("\n");  // Add line-break
                        }
                        else if (tag.equals("mntninfohght")){
                            buffer.append("Mountain_Height : ");
                            xpp.next();
                            buffer.append(xpp.getText()); // Add to StringBuffer from text of "mntninfohght" element
                            buffer.append("\n");  // Add line-break
                        }
                        else if (tag.equals("mntninfopoflc")){
                            buffer.append("Mountain_Address : ");
                            xpp.next();
                            buffer.append(xpp.getText()); // Add to StringBuffer from text of "mntninfopoflc" element
                            buffer.append("\n");  // Add line-break
                        }
                        break;

                    case XmlPullParser.TEXT:
                        break;

                    case XmlPullParser.END_TAG:
                        tag = xpp.getName(); // Get Tag Name

                        if (tag.equals("item")) buffer.append("\n"); // End First Search result.. line-break
                        break;
                }
                eventType = xpp.next();
            }
        } catch (Exception e){
            e.printStackTrace();
        }
        buffer.append("Parsing Finish...\n");
        return buffer.toString(); // Return String Object of StringBuffer
    }
}

File: GraduationProject/app/src/main/java/com/example/graduationproject/MainActivity.kt
package com.example.graduationproject

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.EditText
import android.widget.SearchView
import android.widget.TextView
import android.widget.Toast
import androidx.viewpager.widget.ViewPager
import com.google.android.material.tabs.TabLayout
import com.example.graduationproject.

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    var edit: EditText? = null
    var text: TextView? = null

    // 자신의 서비스키 입력
    var key =
        "LHSwFqTiNHfQEQw4PRhQhq4kY40LwyN9%2Fx4Q9NpOWQr8Dqqql2i630psVHxI4l7Ha37sT292A6M5SR1NCb41rQ%3D%3D"

    //String address = "http://openapi.forest.go.kr/openapi/service/trailInfoService/getforeststoryservice";
    var data: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val sectionsPagerAdapter = SectionsPagerAdapter(this, supportFragmentManager)
        val viewPager: ViewPager = binding.viewPager
        viewPager.adapter = sectionsPagerAdapter
        val tabs: TabLayout = binding.tabs
        tabs.setupWithViewPager(viewPager)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_main , menu)
        val menuItem = menu.findItem(R.id.action_search)
        val searchView = menuItem.actionView as SearchView
        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
            override fun onQueryTextChange(p0: String?): Boolean {
                TODO("Not yet implemented")
                //return true
            }

            override fun onQueryTextSubmit(p0: String?): Boolean {
                TODO("Not yet implemented")
                //return true
            }
        })
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.action_search -> {
                val toast = Toast.makeText(this, "산을 검색해주세요", Toast.LENGTH_SHORT)
                toast.show()
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }
}